devDocs/startupShutdown.md

```diff
@@ -0,0 +1,179 @@
+
+Steps:
+ 1. Press (or emulate) Ctrl+Alt+N, observe NVDA starts up
+
+Variation:
+- At step 1. A version of NVDA is already running. Observe running version exists before installed version starts up.
+
+### Windows Sign-in screen, automatic start
+Prerequisites:
+ - NVDA installed
+ - Enable "Use NVDA during sign-in"
+
+Steps:
+ 1. Sign out (not lock) Windows
+ 1. Observe NVDA announces the Windows sign-in screen
+
+### UAC, automatic start
+Prerequisites:
+ - NVDA installed
+
+Steps:
+
+### Windows Successful sign-in, automatic start
+Prerequisites:
+ - NVDA installed
+ - Enable "Start NVDA after I sign in"
+
+Steps:
+ 1. Start Windows
+ 1. Sign in
+ 1. Observe NVDA starts
+
+### Running the *.exe
+
+Steps:
+ 1. Press `win+r`
+ 1. Enter <path to nvda.exe>
+ 1. Press enter
+ 1. Observe NVDA starts
+
+Variation:
+- using an installer (launcher)
+   -  eg: `C:\Users\username\Downloads\nvda_2021.1.exe`
+- using an installed copy
+   - just type `nvda` in place of the .exe
+- using a portable copy
+   - find and use the path to `nvda.exe`, located within the portable copy directory
+   - the installer allows you to create an installed copy and a portable copy
+
+### Running from source (runnvda.bat)
+Prerequisites
+- clone project and build NVDA (see [project readme](https://github.com/nvaccess/nvda/blob/master/readme.md#getting-the-source-code)).
+
+Steps:
+ 1. Run `runnvda.bat` from cmd
+ 1. Observe NVDA starts
+
+### An input gesture to restart
+
+Prerequisite:
+- Input gesture to restart is assigned to a key
+
+Steps:
+ 1. Press (or emulate) the input gesture
+ 1. Observe that NVDA exits
+ 1. Observe that a new instance is started
+
+## Technical notes
+
+These notes are aimed at developers, wishing to understand technical aspects of the NVDA start and exit.
+
+1. No more than one NVDA process instance should be running time. Interactions with itself could cause severe issues, some (non-exhaustive list) examples of sub-systems where this would be a problem:
+   - NVDA config files
+   - Global (OS level) keyboard hook
+   - Changed / incompatible in-process code
+2. As such, we want to be able to detect running instances, cause them to exit, and confirm they have exited.
+
+### Exit hooks/triggers
+
+There are 3 ways that NVDA receives a request to exit:
+
+- From internally calling [triggerNVDAExit](#When-exiting-from-triggerNVDAExit)
+- Receiving [WM_QUIT](#When-exiting-from-WM_QUIT) Windows message
+- Receiving [wx.EVT_END_SESSION](#When-exiting-from-wxEVT_END_SESSION) due to Windows session ending
+
+### When exiting from `triggerNVDAExit`
+* Called from within NVDA.
+* A function in the core module
+* Only executes the code once, uses a lock and flag to ensure this
+* Uses a queue on the main thread to queue a safe shutdown
+* Once the queued shutdown starts:
+    1. the updateCheck is terminated
+    1. watchdog is terminated
+    1. addons and the brailleViewer are terminated, so we can close all windows safely

```

Comment:
Looking at the current code in `core.py` this seems not to be entirely true since only globalPlugins are  terminated. When reading this my expectation would be that all add-ons are terminated and code which does not belong into an add-on i.e. global plugins in the scratchpad is not both of these are not true since as established above only global plugins are terminated and we do not differentiate between one in scratchpad and one bundled in add-ons. Another question which someone reading this may ask is "What does it mean to terminate an add-on?" There is no class from which all add-ons should inherit and `addonHandler`.terminate` is a no-op.

Reply:
```suggestion
    1. globalPlugins and the brailleViewer are terminated, so we can close all windows safely
```